contracts/ZkSync_SpokePool.sol

```diff
@@ -109,6 +129,17 @@ contract ZkSync_SpokePool is SpokePool {
             WETH9Interface(l2TokenAddress).withdraw(amountToReturn); // Unwrap into ETH.
             // To withdraw tokens, we actually call 'withdraw' on the L2 eth token itself.
             IL2ETH(l2Eth).withdraw{ value: amountToReturn }(withdrawalRecipient);
+        } else if (l2TokenAddress == address(usdcToken)) {
+            if (_isCCTPEnabled()) {
+                // Circle native USDC via CCTP.
+                _transferUsdc(withdrawalRecipient, amountToReturn);
+            } else if (address(zkUSDCBridge) != address(0)) {
+                // Matter Labs custom USDC bridge for Circle Bridged (upgradable) USDC.
+                zkUSDCBridge.withdraw(withdrawalRecipient, l2TokenAddress, amountToReturn);
+            } else {
+                // Fall back to the regular ERC20 bridge.
+                zkErc20Bridge.withdraw(withdrawalRecipient, l2TokenAddress, amountToReturn);
+            }
```

Comment:
This is a pretty clunky construction, but we need it for the zkSync SpokePool, where we have regular zkSync bridged USDC and Circle native USDC, but no Circle bridged (upgradable) USDC. We basically have duplicate calls to the `zkErc20Bridge` here. It'd be nice to avoid that if we can find an alternative construction.

Reply:
The alternative option is to set `usdcToken` to `0x0` on zkSync, in which case we'd never have any special handling for zkSync. I kinda prefer that in terms of code clarity, but it imposes more requirements on the configuration of the SpokePool.

Reply:
I opted for the 2nd option and have added extra comments to the deployment script.